x-pack/plugins/fleet/server/services/security/message_signing_service.test.ts

```diff
@@ -120,7 +120,24 @@ describe('MessageSigningService', () => {
         passphrase: expect.any(String),
       });
 
-      expect(rotateKeyPairResponse).toEqual(true);
+      expect(rotateKeyPairResponse).toEqual(undefined);
+    });
+
+    it('does not generate key pair on rotation if no key pair exists', async () => {
+      mockCreatePointInTimeFinderAsInternalUserOnce([keyPairObj]);
+      // mock delete to return false
+      jest
+        .spyOn(messageSigningService, 'removeKeyPair' as any)
+        .mockReturnValue(Promise.resolve(false));
+
+      const rotateKeyPairResponse = await messageSigningService.rotateKeyPair();
+
+      // can't delete key pair if none exists
+      expect(soClientMock.delete).not.toBeCalled();
+      // doesn't create new key pair
+      expect(soClientMock.create).not.toBeCalled();
+      // returns undefined
+      expect(rotateKeyPairResponse).toEqual(undefined);
```

Comment:
Since this function is a void now, we shouldn't expect it to return anything here right?

Reply:
Oh yeah! Will clean up the test

Reply:
4a1f8643cc86c825893fca11443e554ce38a04b4